sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -491,13 +513,392 @@ function main() {
+
+```ts
+import { ServiceBusClient } from "@azure/service-bus";
+
+async function main() {
+  const browserCredential = new BrowserCredential(clientId, location.origin);
+
+  await browserCredential.prepare();
+
+  if (!browserCredential.isAuthenticated()) {
+    await browserCredential.loginRedirect("https://servicebus.azure.net/.default");
+  }
+
+  const client = new ServiceBusClient(serviceBusEndpoint, browserCredential);
+}
+```
+
+### Authenticating with Key Vault Certificates
+
+Azure Key Vault supports creating secure certificates that can be used to authenticate Azure SDK clients.
+
+There are different ways to create Key Vault Certificates. For example, through the Azure CLI: [Quickstart: Set and retrieve a certificate from Azure Key Vault using Azure CLI](https://docs.microsoft.com/azure/key-vault/certificates/quick-create-cli).
+
+Once you have a certificate, you may export the certificate with the Azure CLI following the steps at [Export certificates from Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/how-to-export-certificate?tabs=azure-cli).
+
+You can also export your certificate through the portal by going to your Key Vault, going to a specific certificate, then downloading the certificate in PFX/PEM format.
+
+Once you have a Key Vault certificate downloaded, go to Azure Active Directory, find the Enterprise app you want to authenticate with, go to `Certificates & secrets` and then upload the certificate.
+
+After that, you'll be able to authenticate by pointing the `@azure/identity`'s `ClientCertificateCredential` to the path where your PEM certificate is, as follows:
+
+```ts
+const credential = new ClientCertificateCredential(
+  "<your-tenant-id>",
+  "<your-client-id>",
+  "<the-path-to-your-certificate-in-PEM-format>"
+);
+```
+
+### Rolling Certificates
+
+Long-running applications may need to roll certificates during process execution. At the moment, the `ClientCertificateCredential` does not support certificate rotation (the credential treats the certificate provided as immutable). Therefore, clients constructed with a `ClientCertificateCredential` using a particular cert would fail to authenticate requests after that certificate rolls and the original is no longer valid.
+
+However, if an application wants to roll this certificate without creating new service clients, it can accomplish this by creating its own `TokenCredential` implementation, which wraps the `ClientCertificateCredential`. Implementing this custom `TokenCredential` would somewhat depend on how the application handles certificate rotation.
+
+#### Explicit rotation
+
+If the application gets notified of certificate rotations and can directly respond, it might choose to wrap the `ClientCertificateCredential` in a custom credential which provides a means for rotating the certificate.
+
+> You'll need to install the [@azure/core-auth][core_auth] package for this sample.
```

Comment:
We were thinking of a pre-requisites section at some point in this same PR, but there are several questions that aren’t too simple to answer:

- What is needed from the perspective of somebody new to TypeScript? We do have some of that information in our main Identity readme, should we have something similar here?
- Since the dependencies are not exactly unifiable for all of these samples, a pre-requisites section may add confusion regarding what is needed when.
- People may share and jump directly to some specific sample in this document, so local context may help better than a pre-requisites section.

What do you think? I want to provide this so that you can help me think what makes sense to you. I’ll take what you decide for sure.

Reply:
> What is needed from the perspective of somebody new to TypeScript? We do have some of that information in our main Identity readme, should we have something similar here?

You could list TypeScript as a prerequisite and link to https://www.typescriptlang.org/download. That information is probably better off in the main README. No need to duplicate the content then.

What I meant by adding a Prerequisites section was to add a heading before something like the following:

```
> You'll need to install the [@azure/core-auth][core_auth] package for this sample.
```

This example would become:

```
**Prerequisites**

Install the [@azure/core-auth][core_auth] package.
```

Reply:
Oh ok ok, I can do that!